AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasNodeInspector.py

```diff
@@ -0,0 +1,109 @@
+            Number: float
+            Transform: list[math.Vector3, math.Vector3, float]
+            Vectors use corresponding math library vector
+
+        params variable_name: the name of the variable that's selected
+        params variable_type: the type of the variable being configured.
+        params new_value: the new value you want to initialize the variable with
+
+        returns None
+        """
+        #todo
+        #flesh out the rest of the switch
+        #add better verification that the client is trying to modify the correct type of variable
+
+        variable_qobject = self.node_inspector.findChild(QtWidgets.QFrame, f"{variable_name}")
+
+        match variable_type:
+            case "Boolean":
+                assert type(new_value) == BooleanCheckBoxValues, f"New value: {new_value} is invalid for this variable type:{variable_type}"
+                self.__change_variable_value_Boolean(variable_qobject, new_value)
+            case "Color", "String":
+                assert type(new_value) == str, f"New value: {new_value} is invalid for this variable type:{variable_type}"
+                self.__change_variable_value_string(variable_qobject, new_value)
+
+    def __change_variable_value_Boolean(self, variable_qobject: QtWidgets.QFrame, new_value: BooleanCheckBoxValues) -> None:
+        """
+        helper function for the change_variable initial_value branch (boolean variable type)
+        """
+        check_box = variable_qobject.findChild(QtWidgets.QCheckBox, "")
+        new_value_to_bool = bool(new_value)
+
+        if new_value_to_bool != check_box.isChecked():
+            check_box.click()
+
+
+    def __change_variable_value_string(self, variable_qobject: QtWidgets.QFrame, new_value: str) -> None:
+        """
+        helper function for the change_variable_initial_value branch(color, string variable type)
+
+        params variable_qobject: the qframe we need to drill into
+        params new_value: the value we are replacing the old value with
+
+        returns None
+        """
+
+        qline_edit = variable_qobject.findChild(QtWidgets.QLineEdit, "")
+        helper.wait_for_condition(lambda: True is False, WAIT_TIME_SEC_1)
+
+        qline_edit.setText(f"{new_value}")
+
+
+    def change_variable_initial_value_source(self, value_source: BooleanCheckBoxValues):
+        """
+        Function for changing the scope of a variable. if you change the value source to 'from component' the variable
+        will be exposed the script canvas component.
+
+        Note: This function assumes the desired node is already in focus and visible to the node inspector
+
+        params value_source: the index of the selection from the combo box Off = from graph On = from component
+
+        """
+        # local variables aren't all necessary but extracting them makes getting to the data easier to understand
+        variable_property_editor_frame = self.node_inspector.findChild(QtWidgets.QFrame, "PropertyEditor")
```

Comment:
Could this be moved into a class variable that only needs to be initialized once?

Reply:
The qt hierarchy changes and refreshes as you click on different variables and nodes. I wouldn't trust that qframe to be static even if it's always there for different variables.